Add timezone configuration

Angel-Padilla opened the pull request:
Two configurations were added to the CONFIG_USER: section of the
MeticulousConfig dict

 - Timezone_sync: This can be 'manual' or 'automatic' which indicates
if the machine should update the timezone automatically or not
 - timezone: this config holds the timezone name currently active in the
system

The TimezoneManager class is created to handle changes in the timezone
configurations, the class implements the following methods

 - get_organized_timezones: generates a json file mapping the timezones and cities with countries for UI
 - validate_timezones_json: checks for the tz generated json presence, or generates it if missing
 - get_system_timezone: retrieves the current system
 - set_system_timezone: sets the system timezone using `timedatectl`
 - update_timezone: updates system timezone and updates the config file
 - init: validates json file and updates config file with system timezone if needed
 - get_UI timezones: provides the tz json file as a dictionary
 - request_and_sync_tz: request the current timezone from `analytics.meticulousespresso` and updates the tz
 
Added TIME_ZONE and TIMEZONE_SYNC configurations to the /settings endpoint post request

Added TimezoneUIProvider class to provide to the dial app the requested filtered list
of countries/cities to be chosen as timezone selectors
the endpoint for this request is `/timezones/(.*)`

Fix:  endpoint capture clause added to get the setting name to update

Octavio2121 committed:
Some endpoints related to the OS update are added to the backend API

Octavio2121 committed:
Remove dbus messages

Octavio2121 committed:
Remove dbus messages and notify

Octavio2121 committed:
Remove dbus messages, add notify

Mimoja committed:
profile: create deepcopy when processing variables

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
docker: run with normal user inside the container

Mimoja committed:
sounds: disable all default sounds besides notifications and heating end

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
api: fix shot history api endpoint

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
history: write full profile to shot file

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
ci: install system dependencies

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
ShotManager: add id field to shot

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
history: Add sqlite and sqlite-zstd dependencies

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
profile: always expand image filename to full url on save

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Commit:
Fix: Incorrect character for new workflow

We have been sending 'hash ' when transmitting the
hash. It has worked so far due to the previous
method. With the refactor on how data is parsed on
the ESP32, consistency is needed in those messages.
'hash ' has been changed to 'hash,'.

Mimoja committed:
docker: add ignore files

Signed-off-by: Mimoja <git@mimoja.de>

Mimoja committed:
python: move bless and pyImprov to pypi hosted packages

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
Revert "sound: relace playsound with mpv"

This reverts commit 19204cdd85fb3b4bb57f7544528825b091944fb0.

Mimoja committed:
fix broken formating

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
profiles: remove legacy italian support

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
ci/cd: add container build workflow

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
ci/cd: Build docker container on arm64 as well

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
back, profile: Allow profile hovering via socket.io message

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
Use sentry for crash tracking

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
machine: autoformat machine.py

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
machine: cleanup the firmware upgrade notification

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
notifications: create new thread when updating from the API

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
imager: use fallback file in case emmc is already imaged

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
images: format flashing progress a bit nicer

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
imager: add notification image

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
notifications: dont delete out of bounds on update

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
imager: fix gpt and meticulous-user partitions just in case

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
imager: dont hang if fdisk or mkfs fails

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
imager: unmount all partitions before and after writing"

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
update profile schema

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
profiles: Add default profiles api and submodule

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
fix(history): write time as unix timestamp to disk

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
ShotManager: put raw shot files into a proper subfolder

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
history: implement shot database

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
api: add new history endpoints for database supported requests

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
flake8: add flake8 condfig

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
api, history: Add last and current shot endpoints

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
history: reset last_shot only after the new shot has been sucessfully writen

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
Add pytz to requirements.txt

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
run black formater on the entire codebase

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
flake8: fix all flake8 errors

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
ci/cd: add flake8 and black validation on push

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
ci/cd: remove unused pytest command for now

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
history: dont scan the disk for shotfiles on startup

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
api: remove leftover test HTML page

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

MiguelOsorno committed:
feat(profiles): add logic to generate ramdom accent color

Commit:
fix(emit): fix emit to the same device

Mimoja committed:
api/history: return the complete history, all shots on get request

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
history: add database statistics

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
shotDB: cleanup logging

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
imager: warn if the image is bigger than the target disc

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
shotDB: annotate dump_data field for pydantic

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
profle: ensure the random colors are always 6 characters long

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
shotDB: disable compression as we are not supporting sqlite modules in our python3.12

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
imager: fix init

Mimoja committed:
wifi: mix emulated offline configuration

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
Revert "shotDB: disable compression as we are not supporting sqlite modules in our python3.12"

This reverts commit 53d91ae62e61f041a094424f0ff8604339d570b4.

Mimoja committed:
data: Fix pressure_sensor typo

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
web/debug: add sensor value plotting

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
web/debug: reformat table

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
web/debug: turn sensor flexbox into a grid

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
emulated: dont print firmware version every few seconds

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
wifi/tryAutoConnect: create asyncio loop for wifi thread

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
wifi: fix asyncio loop warning

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
sentry, config: add the config as sentry context

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
main: move the backend logic into its own file and only init sentry in the entrypoint

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
shotDB: store full display object instead of only the image

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

AUOR17 committed:
feat(api/profiles.py): add 'legacy' endpoint.

Introduce a 'legacy' endpoint allowing the submission of partial JSON nodes without a complete profile.
This is controlled by the 'legacy_json' config option, which defaults to false.
When set to true, users can access the endpoint at /api/v1/profile/legacy.

- Add 'legacy' endpoint for partial JSON submission
- Introduce 'legacy_json' config option, defaulting to false
- Implement logic to bypass profile completeness check when 'legacy_json' is true"

AUOR17 committed:
Reformat

Mimoja committed:
api/history/debug: use correct path for debugg logs

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
default_profiles: Fix lever profiles blooming stage

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
debug_shots: Add profile time in ms

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Angel-Padilla committed:
fix: reset action

When receiving a reset action, the API was forwarding the action to the
ESP32, but it cannot reset itself, so the action was not being executed
even is the response was successful, this commit separates the valid
actions to ESP_ACTIONS and BACKEND_ACTIONS, so the reset action is not
forwarded to the ESP32 but executed by the backend itself. The new
ESP_ACTIONS are: "start", "stop", "tare"
while the BACKEND_ACTIONS are: "reset"

AluUriel committed:
fix typo

Mimoja committed:
history: remove data column from database and read from disc instead

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
machine: no longer print sensor messages during brew

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
Add memray option for local running

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
profile: set a dummy profile is deprecated nodeJSON is used

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
wifi: limit the access points names length

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
api: fix emulation import path

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
debug-ui: add all missing state fields

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
shotDB: add safeguards against shots without a profile

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
shotDB: dont assume stages, variales or previous authors to always exist

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
shotDB: fix typo

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
shotDB: fix profile unpacking

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
debug: reduce the debug shot compression to not eat up all CPU

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
debug: fix time spend calculation

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
log: remove file logging completely to investigate a potential memory leak

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
history: explicitly delete the shot and debug shot json

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
log: fix leftover file rotation logic

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
threading: Create NameThread wrapper class forcing thread nameing and preparing monitoring

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
NamedThread: add missing file

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
machine: emit sensor data immediatly once it is available from the machine thread

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
NamedThread: reduce logging level

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
Fully disable all calls to playsound while we debug an ALSA issue

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
chore: fix liniting errors

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
Revert "machine: emit sensor data immediatly once it is available from the machine thread"

This reverts commit 7ac68979b72a75b6100a99ddf40c5db076169678.

Mimoja committed:
Revert "machine: emit sensor data immediatly once it is available from the machine thread"

This reverts commit 7ac68979b72a75b6100a99ddf40c5db076169678.

Octavio2121 committed:
Minor change, the file path to update_os.py was changed

Mimoja committed:
api/profile: move legacy profile config into the user accessible section

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
machine: Initialize the Machine status as IDLE

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

AUOR17 committed:
Fix legacy jason

AUOR17 committed:
Erase unused module

Mimoja committed:
machine: mock heat_on_boot logic

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Angel-Padilla committed:
feat: add bh_current handle support at SensorData class

Mimoja committed:
ble: build custom announcement for connection data

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Angel-Padilla committed:
feat: add LAM_temp support at SensorData class

Mimoja committed:
machine: parse extraCommits into int

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
wifi:_update ble advertisment on successfull connection

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
machine: parse new setpoint states

The firmware is now able to send the current setpoints as well.
To be able to make use of it we are extending the esp data parser and
to be able to handle these new fields.

The socket io message was extended by a setpoint object

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
debug: add setpoints to ui

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
debug: rename fields to be more expressive

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
debug: fix missing setpoint handling

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
debug: use correct setpoint field

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
debug: add setpoints to the graph

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
shotDB: fix broken scoping

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
history: add setpoints to shot files

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
history: remove temperature data

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
history/zstd: Use basehandler default headers

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
remove legacy profile leftovers

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
ble: fix logging messages

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
emulation: add setpoint data

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
sio: announce setpoints as lowercase

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
remove more leftovers

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Angel-Padilla committed:
fix weird snake case variable naming

AluUriel committed:
fix: flash esp32 when data is not receievd from it

AluUriel committed:
Fix: Skipping retracting stage

Mimoja committed:
wifi: allow connecting to different wifi types

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
emulation: dont fail at parsing the emulation data for now

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
ota: allow setting of the update channel

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
api/settings: fix settings typecheck

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
profile: measure pre-processing times

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Angel-Padilla committed:
IPC:first working commit of dbus monitoring

Angel-Padilla committed:
IPC:first working commit of dbus monitoring

Angel-Padilla committed:
notifications: add notifications to the progress, errors and completion of rauc updates

Mimoja committed:
machine: add serial number setting endpoint

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
nit: fix linting

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
machine: no longer randomly generate the serial

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Commit:
Feat: Info is being stored in the nvs of esp32
Relevant info as Color, Serial_number, Batch_number and Build_date will is now stored
within the esp32 nvs

Mimoja committed:
machine: fix parsing of ESPInfo message

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Angel-Padilla committed:
fix notification:

Progress notifications didnt update if the user didnt aknowledge them. Prgress notifications are dismissed when the "Completed" or any "Error" notification are sent

Angel-Padilla committed:
refactor: rauc dbus

fix MeticulousLogger.getLogger()

Mimoja committed:
machine: save the serial and related informations once recieved from the firmware

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

oscarlr16 committed:
feat(socket): add handler for setting heater timeout

Introduce functionality to set heater timeout through socket events.
This enables clients to control heater timeout dynamically via the 'set_heater_timeout' event.

- Add 'set_heater_timeout' event handler in backend.py
- Log the timeout value received from the client
- Format and send timeout data to the machine

Mimoja committed:
machine: Add esp based notification logic

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
api/action: Allow for scale master calibration

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
machine: fix acaia message handling to properly parse the message

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Angel-Padilla committed:
refactor(AsyncDBUSClient): added start function member to class

Angel-Padilla committed:
rename(dbus_callbacks): rauc_dbus.py renamed to dbus_callbacks.py

Angel-Padilla committed:
feat(backend.py):dbus object configured to track signals and rauc interface attributes, as well as thumbdrives notifications

Angel-Padilla committed:
styling():fix style according to flake8

Angel-Padilla committed:
subscribe dbus object to rauc download bundle progress signal

oscarlr16 committed:
feat(machine): add support for processing heater timeout info

    Introduce HeaterTimeoutInfo class for handling microcontroller timeout data.
    Add methods to parse arguments and convert data to a dictionary. Update Machine
    class to process and log heater timeout information from the microcontroller.

    - Create HeaterTimeoutInfo class with parsing and dictionary conversion methods
    - Update Machine class to handle HeaterTimeoutInfo data
    - Add error handling for invalid heater timeout arguments
    - Log HeaterTimeoutInfo processing results

Angel-Padilla committed:
run black formatter

Mimoja committed:
profile: add 'preparing' stage to avoid flickering when purging

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Angel-Padilla committed:
refactor(dbus_monitor): implementation of DBusMonitor class

the class dbusCallbacks that was only a namespace for the callbacks used by the
dbus_object now also wraps the AsyncDBUSClient class and changed its name to
DBusMonitor

Angel-Padilla committed:
chore(dbus_client): add SPDX license header and reference to source code

Angel-Padilla committed:
refactor(dbus_monitor): use the class directly instead of intantiating an object

oscarlr16 committed:
refactor(heater): reorganize heating timeout handling in settings

Refactored code to properly handle heating timeout via user settings and ensure secure validation.
Improved code organization and added logging for invalid timeout values.

- Move  to user settings in
- Update  to process heating timeout with validation
- Add logging and error handling for incorrect timeout values
- Refactor imports and structure in

Angel-Padilla committed:
feat(api/machine.py): Added classes that hold and report the status of the OS update

Angel-Padilla committed:
feat(dbus_monitor): implement OS Update reporting on dbus_monitor.py

Angel-Padilla committed:
configure updateOSStatus socketIO

Angel-Padilla committed:
refactor(machine.py):
 data dictionary now is attribute of the UpdateOSStatus class instead of
local variable of sendUpdateStatus, this allows us to keep its data at hand to later
access it in a REST endpoint

Angel-Padilla committed:
feat(machine.py): implementation of OS_update_status REST endpoint

an endpoint at machine/OS_update_status is implemented to get the latest status of the
update beside the socketio communication

Angel-Padilla committed:
run formatter

Angel-Padilla committed:
feat(dbus_monitor): replace notifications with OSUpdateStatus'

black formatting

Angel-Padilla committed:
chore(api/machine.py): round update os progress to integer

Angel-Padilla committed:
feat(OTA): keep recovery updates using notifications

When an update is triggerd by plugging an USB, meaning it is a recovery
update, the backend will still send the notification to the dia besides the
socketIO notifications.

To identify that the update comes from the USB, the backend will listen for
a dbus signal `RecoveryUpdate` from the interface `com.Meticulous.Handler.MassStorage`

Mimoja committed:
requirements: downgrade pygobject to be compatible with the debian version

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

oscarlr16 committed:
feat(graph): add band heater current display in UI

Enhance the UI by adding band heater current data display.
Updated the HTML and JS files to support visualization of band heater current.

- Add 'BH Current' label and data field in
- Update  to handle
- Add  to  and data update logic
- Assign white color for  in the color mapping

AUOR17 committed:
feat(): add motor temperature data for REL testing

Add configuration option to return thermistor temperature data if enabled.
If the option is disabled, return 0. Thermistor data added to sensor group
and a new 'accessory data' group created in the JSON output.

- Add config option to toggle thermistor data return
- Update sensor group to include thermistor data
- Create new 'accessory data' group in JSON for additional info

Angel-Padilla committed:
refactor(dbus_monitor): round progress of os update download

Angel-Padilla committed:
run black formatter

Angel-Padilla committed:
fix(dbus_monitor): remove all notifications when the OS update is not recovery

Angel-Padilla committed:
fix(dbus_monitor): format according to flake8

Mimoja committed:
ble_gatt: fix wrong call to WifiManager

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
wifi: fix invalid credential type on connect

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Angel-Padilla committed:
refactor(api/machine.py): Added classmethod to_json() to UpdateOSStatus class

the class method to_json() will return a json containing the data from
`last_progress`, `last_status` and `extra_info` correctly formatted to be
consumed

oscarlr16 committed:
feat(machine): emit heater status when preheat is complete

Add logic to emit 'heater_status: off' when preheating completes.
This improves tracking of heater status in real-time.

- Emit 'heater_status: off' if preheat_remaining equals 0
- Log heater status emission for tracking

AUOR17 committed:
Remove simulate temp function

Angel-Padilla committed:
feat(dbus_monitor): add d_bus signal suscription to monitor recovery failures

Some times when the recovery update fails, a fail message is not reported by
rauc, so the installing os notification is not dismissed and replaced with
an error notification, making it look like it gets stuck.

Now a dbus signal sent from the media_rauc_install.sh script to let know an
error happend, and not only relies on the rauc Progress dbus property

oscarlr16 committed:
feat(machine): emit preheat_remaining every second during preheat

Ensure consistent emission of heater preheat status by sending 'preheat_remaining' every second until the preheat completes.

- Add emission of 'heater_timeout_info.preheat_remaining' every second
- Remove redundant emission of 'heater_status: off' when preheat is 0
- Retain logging of 'heater_status: off' for visibility

Mimoja committed:
backlight: Add backlight controller and API

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
backlight: adjust in its own thread

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
backlight: reorder api

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
backlight: move thread logic into the backlight controller

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
backlight: re-add accidentally deleted functions

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
backlight: up the min_brightness to 1500

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Commit:
Feat: Predictive weight enable
With the firmware update, the weight prediction has been enabled.
However, JSONs generated in the backend are hardcoded to use
"Weight Raw" as a source. To ensure that this option is widely
used, the source has been changed to "Weight Predictive." Further
discussion should be held to enable the use of raw or predictive
triggers based on user preferences.

oscarlr16 committed:
fix(heater): prevent redundant 'heater off' logs

Ensure the 'Heater_status: off' message is logged only once when the heater's preheat time reaches zero. This avoids unnecessary log entries for repeated heater status updates.

- Add  variable to track heater's state
- Check previous value before logging 'Heater_status: off'
- Update value of  after each status check

Commit:
fix(esp_serial/data.py): replace "nan" values with string literal "NaN" when sending sensor data through Socket.io

Commit:
fix(esp_serial/data.py): resolve issues with code formatting

oscarlr16 committed:
fix(ble): prevent connection drops during wifi provisioning

when wifi is provisioned via app, ble connection was being interrupted before
response could be sent back to client. this fixes the issue by:

- add check_client_connections() to verify active ble connections
- make update_advertisement() async to properly handle connection checks
- add safe update helper method in wifi manager
- update all advertisement calls to use new helper

Mimoja committed:
add temperature to the setpoints

Signed-off-by: Mimoja <mimoja@meticuloushome.com>

Mimoja committed:
history: allow to query for profile ids directly

Signed-off-by: Mimoja <git@mimoja.de>

oscarlr16 committed:
feat(audio): optimize sound playback system with new implementation

Replace inefficient playsound.py module with new custom play_sound.py implementation
featuring proper resource management and thread synchronization. Updates include:

- Implement new SoundPlayer class with explicit cleanup methods
- Add thread-safe playback operations using locks
- Switch from blocking poll to event-based message handling
- Enable proper playback of system sound events

Angel-Padilla committed:
chore: cleanup code that referenced refactored code

Angel-Padilla committed:
feat: update Docker's Debian base image from bullseye to bookworm version

Angel-Padilla committed:
fix: start the dbus service before launching the backend service

The base image does not have the dbus nor the bluez packages installed
which are required for rauc notifications as well as bluez interface collection

both packages are installed in the Docker imaged build process.

A new entry point is defined to start the dbus service before launching the
backend service.

There is still an issue with the bluetooth, it cannot be activated so the Gatt
server crashes

oscarlr16 committed:
fix(serial-number): generate random serial if missing

Ensure every machine has a valid serial number by generating one when
none exists or if it's empty. Saves the new serial number to the configuration.

- Add  method in Machine class
- Update MachineInfoHandler to check for missing or empty serial
- Generate random serial and save to config if none is set
- Modify response to include generated serial

Angel-Padilla committed:
feat(config.py): add Timezone configs to MeticulousConfig dict

2 configurations were added to the CONFIG_USER: section of the
MeticulousConfig dict

 - Timezone_sync: This can be 'manual' or 'automatic' which indicates
if the machine should update the timezone automatically or not
 - timezone: this config holds the timezone name currently active in the
system

Angel-Padilla committed:
feat(timezone_manager): add timezoneManager

The TimezoneManager class is created to handle changes in the timezone
configurations, the class implements the following methods

 - get_organized_timezones: generates a json file mapping the timezones and cities with countries for UI
 - validate_timezones_json: checks for the tz generated json presence, or generates it if missing
 - get_system_timezone: retrieves the current system
 - set_system_timezone: sets the system timezone using `timedatectl`
 - update_timezone: updates system timezone and updates the config file
 - init: validates json file and updates config file with system timezone if needed
 - get_UI timezones: provides the tz json file as a dictionary
 - request_and_sync_tz: request the current timezone from `analytics.meticulousespresso` and updates the tz

Angel-Padilla committed:
feat(backend.py): initialize the TimezoneManager

Angel-Padilla committed:
feature(settings.py): add tz endpoints configuration

Added TIME_ZONE and TIMEZONE_SYNC configurations to the /settings endpoint post request

Added TimezoneUIProvider class to provide to the dial app the requested filtered list
of countries/cities to be chosen as timezone selectors
the endpoint for this request is `/timezones/(.*)`

Angel-Padilla committed:
fix:  endpoint capture clause added to get the setting name to update

oscarlr16 committed:
feat(data): only emit temperature when machine status is heating

Mimoja reviewed (changes requested):
Looks good, only the request to the server can timeout and hang the entire machine, Lets prevent that please :D

Angel-Padilla committed:
fix(settings.py): Multiple jsons sent as HTTP response

When a post request coming to the /settings/ endpoint aims to change the
time_zone or timezone_sync configs, the HTTP response contained 2 jsons
one containing the response for if the change was successfull followed by
the dictionary dump of the user config section on the meticulousconfig.
The later was removed as is not expected back

fix: change setting_name to setting_target

Angel-Padilla committed:
chore(timezone_manager): remove double imports

oscarlr16 committed:
feat: add flow_from_scale property to ShotData

Introduce flow_from_scale property to enhance data tracking and logging.

- Add flow_from_scale as a property in ShotData
- Parse flow_from_scale from args[11] in ShotData parsing logic
- Include flow_from_scale in ShotData JSON serialization
- Update ShotManager to include flow_from_scale in shot data output

oscarlr16 committed:
refactor(sensor-data): remove unused valve sensor

The 'valve' sensor field was removed as it is no longer in use.
This simplifies the code and reduces unnecessary data handling.

- Remove 'valve' attribute from SensorData class
- Remove 'valve' processing in data parsing methods
- Remove 'valve' from serialized output

oscarlr16 committed:
feat(debug): add 'flow_from_scale' to debug data output

Enhance debug data by including 'flow_from_scale' in the output.
This provides additional context for shot debugging and logging.

- Add 'flow_from_scale' to the debug data dictionary in

Mimoja committed:
settings: add idle screen config

Signed-off-by: Mimoja <git@mimoja.de>

Mimoja committed:
profile_converter: Add config to skip stages

Signed-off-by: Mimoja <git@mimoja.de>

Angel-Padilla committed:
make timezone request async

Angel-Padilla committed:
chore: reduce time out time for tz http request

Angel-Padilla committed:
chore: remove commented code

Angel-Padilla committed:
chore: format code

Angel-Padilla committed:
feature: update requirements

AUOR17 committed:
fix: water detection water detection stage

There was a bug in the init node of water detection stage

oscarlr16 committed:
refactor(data): remove temperature condition in sensor data

Let the dial app handle when to show temperature in barometer view instead of filtering at data level

oscarlr16 committed:
feat(api): add latest file retrieval endpoint for debug history

- Add get_latest_file method to find most recent file in debug history directory

- Implement handling of latest path in get method to redirect to most recent file

- Sort through date-based folder structure to find latest timestamp

oscarlr16 committed:
refactor: rename flow_from_scale to gravimetric_flow

- Rename flow_from_scale to gravimetric_flow for better clarity and intuitive understanding

- Update sensor dictionary key to 'g' to maintain single-letter convention with other sensors

oblador committed:
fix(history): respect query params in endpoint

oscarlr16 committed:
feat(machine): add mainVoltage to response payload

Includes mainVoltage in the response of the MachineInfoHandler endpoint.
This allows easier access to voltage data, which is useful for future testing.

- Add mainVoltage to the response payload in MachineInfoHandler

oscarlr16 committed:
refactor(api): improve debug file handling with dedicated handler

- Create dedicated LastDebugFileHandler for retrieving most recent debug files
- Leverage existing ShotManager.getLastShot() instead of directory traversal
- Improve naming to clearly indicate debug file context
- Remove redundant latest file logic from ZstdHistoryHandler

Angel-Padilla committed:
refactor: move MACHINE_ALLOW_STAGE_SKIPPING system config to user config as it will now be updated from dial app

Mimoja committed:
sounds/theme: disable startup sound

Signed-off-by: Mimoja <mimoja@mimoja.de>

Octavio2121 committed:
api, ota: add build channel to machine info

Mimoja committed:
ota: Add BUILD_DATE parsing to the UpdateManager

Signed-off-by: Mimoja <mimoja@mimoja.de>

Mimoja committed:
api/machine: Serve Build date

Signed-off-by: Mimoja <mimoja@mimoja.de>

Commit:
Add USB mode settings and API

Signed-off-by: root <root@meticulousDevMimojaFika1.fritz.box>

Mimoja committed:
usb: fix wrong call to enum.value

Signed-off-by: Mimoja <mimoja@mimoja.de>

Mimoja committed:
usb: use enum parsing capabilities to check validity of input

Signed-off-by: Mimoja <mimoja@mimoja.de>

AUOR17 committed:
Purge trigger added to end a profile

Mimoja committed:
usb: dont fail if usb modules is not connected

Signed-off-by: Mimoja <git@mimoja.de>

Mimoja committed:
profile_converter: disable the heating offset

Signed-off-by: Mimoja <git@mimoja.de>

Mimoja committed:
ci/cd: remove stale dev and main branch scripts

Signed-off-by: Mimoja <git@mimoja.de>

Mimoja committed:
ota: repo info summay into a dictionary

Signed-off-by: Mimoja <git@mimoja.de>

Mimoja committed:
debug-ui: stretch the grid a bit to reduce flickering

Signed-off-by: Mimoja <git@mimoja.de>

Mimoja committed:
profile_converter: Reduce max piston position to 77 to ensure we dont flicker on purge startup

Signed-off-by: Mimoja <git@mimoja.de>

Mimoja committed:
profile_converter: Reduce max piston position even furth to 75

Signed-off-by: Mimoja <git@mimoja.de>

Octavio2121 committed:
Merge pull request #175 from MeticulousHome/feat_summary

ota: repo info summay into a dictionary

Angel-Padilla committed:
fix: change setting_name to setting_target

Angel-Padilla committed:
nit fixes:

 - remove TODO comment on async timezone getter function
 - change class name from `TimezoneUIProvider` to `TimezoneUIHandler`

AUOR17 committed:
feat(motor_thermistor): add motor thermistor data

- Add motor thermistor data to the graph
- This is in order to get the temperature in real time of the linear actuator

AUOR17 committed:
Fix typo error
